var/spack/repos/builtin/packages/clingo-bootstrap/package.py

```diff
@@ -123,7 +123,9 @@ def pgo_train(self):
         # Run spack solve --fresh hdf5 with instrumented clingo.
         python_runtime_env = EnvironmentModifications()
         for s in self.spec.traverse(deptype=("run", "link"), order="post"):
-            python_runtime_env.extend(spack.user_environment.environment_modifications_for_spec(s))
+            python_runtime_env.extend(
+                spack.user_environment.environment_modifications_for_specs(s)
+            )
```

Comment:
Can you take the changes from https://github.com/spack/spack/blob/4e2e043237c30b76d034bccefadbff494ae24b59/var/spack/repos/builtin/packages/clingo-bootstrap/package.py?

The loop causes quadratic complexity for a linear operation.

Reply:
Done.